File: Tests/Main Tests/WordAddinVB/SampleControl.vb
Public Class SampleControl
    Implements NetOffice.WordApi.Tools.ITaskPane

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        MsgBox("Hello")
    End Sub

    Public Sub OnConnection(ByVal application As NetOffice.WordApi.Application, ByVal parentPane As NetOffice.OfficeApi._CustomTaskPane, ByVal customArguments() As Object) Implements NetOffice.WordApi.Tools.ITaskPane.OnConnection

        Dim addin As TestAddin = customArguments(0)
        addin.TaskPaneOkay = True

    End Sub

    Public Sub OnDisconnection() Implements NetOffice.WordApi.Tools.ITaskPane.OnDisconnection

    End Sub

    Public Sub OnDockPositionChanged(ByVal position As NetOffice.OfficeApi.Enums.MsoCTPDockPosition) Implements NetOffice.WordApi.Tools.ITaskPane.OnDockPositionChanged

    End Sub


    Public Sub OnVisibleStateChanged(ByVal visible As Boolean) Implements NetOffice.WordApi.Tools.ITaskPane.OnVisibleStateChanged

    End Sub

End Class

File: Tests/Main Tests/WordAddinCSharp/SampleControl.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace WordAddinCSharp
{
    public partial class SampleControl : UserControl, NetOffice.WordApi.Tools.ITaskPane
    {
        public SampleControl()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            MessageBox.Show("Hello");
        }

        void NetOffice.WordApi.Tools.ITaskPane.OnConnection(NetOffice.WordApi.Application application, NetOffice.OfficeApi._CustomTaskPane parentPane, object[] customArguments)
        {
            TestAddin addin = customArguments[0] as TestAddin;
            addin.TaskPaneOkay = true;
        }

        void NetOffice.WordApi.Tools.ITaskPane.OnDisconnection()
        {

        }

        void NetOffice.WordApi.Tools.ITaskPane.OnDockPositionChanged(NetOffice.OfficeApi.Enums.MsoCTPDockPosition position)
        {

        }

        void NetOffice.WordApi.Tools.ITaskPane.OnVisibleStateChanged(bool visible)
        {

        }
    }
}
